audio/pwaudio.c

```diff
@@ -14,26 +14,357 @@
+        *endianness = 1;
+        return AUDIO_FORMAT_U16;
+    case SPA_AUDIO_FORMAT_U16_LE:
+        *endianness = 0;
+        return AUDIO_FORMAT_U16;
+    case SPA_AUDIO_FORMAT_S32_BE:
+        *endianness = 1;
+        return AUDIO_FORMAT_S32;
+    case SPA_AUDIO_FORMAT_S32_LE:
+        *endianness = 0;
+        return AUDIO_FORMAT_S32;
+    case SPA_AUDIO_FORMAT_U32_BE:
+        *endianness = 1;
+        return AUDIO_FORMAT_U32;
+    case SPA_AUDIO_FORMAT_U32_LE:
+        *endianness = 0;
+        return AUDIO_FORMAT_U32;
+    case SPA_AUDIO_FORMAT_F32_BE:
+        *endianness = 1;
+        return AUDIO_FORMAT_F32;
+    case SPA_AUDIO_FORMAT_F32_LE:
+        *endianness = 0;
+        return AUDIO_FORMAT_F32;
+    default:
+        dolog ("Internal logic error: Bad spa_audio_format %d\n", fmt);
+        return AUDIO_FORMAT_U8;
+    }
+}
+
+static struct pw_stream *qpw_stream_new (
+    pwaudio *c,
+    const char *name,
+    struct spa_audio_info_raw *info,
+    int *error)
+{
+    int r;
+    struct pw_stream *stream = NULL;
+    //enum spa_audio_channel position[8];
+
+    pw_thread_loop_lock(c->thread_loop);
+
+    switch (info->channels) {
+	case 8:
+		info->position[0] = SPA_AUDIO_CHANNEL_FL;
+		info->position[1] = SPA_AUDIO_CHANNEL_FR;
+		info->position[2] = SPA_AUDIO_CHANNEL_FC;
+		info->position[3] = SPA_AUDIO_CHANNEL_LFE;
+		info->position[4] = SPA_AUDIO_CHANNEL_RL;
+		info->position[5] = SPA_AUDIO_CHANNEL_RR;
+		info->position[6] = SPA_AUDIO_CHANNEL_SL;
+		info->position[7] = SPA_AUDIO_CHANNEL_SR;
+		break;
+	case 6:
+		info->position[0] = SPA_AUDIO_CHANNEL_FL;
+		info->position[1] = SPA_AUDIO_CHANNEL_FR;
+		info->position[2] = SPA_AUDIO_CHANNEL_FC;
+		info->position[3] = SPA_AUDIO_CHANNEL_LFE;
+		info->position[4] = SPA_AUDIO_CHANNEL_RL;
+		info->position[5] = SPA_AUDIO_CHANNEL_RR;
+		break;
+	case 5:
+		info->position[0] = SPA_AUDIO_CHANNEL_FL;
+		info->position[1] = SPA_AUDIO_CHANNEL_FR;
+		info->position[2] = SPA_AUDIO_CHANNEL_FC;
+		info->position[3] = SPA_AUDIO_CHANNEL_LFE;
+		info->position[4] = SPA_AUDIO_CHANNEL_RC;
+		break;
+	case 4:
+		info->position[0] = SPA_AUDIO_CHANNEL_FL;
+		info->position[1] = SPA_AUDIO_CHANNEL_FR;
+		info->position[2] = SPA_AUDIO_CHANNEL_FC;
+		info->position[3] = SPA_AUDIO_CHANNEL_RC;
+		break;
+	case 3:
+		info->position[0] = SPA_AUDIO_CHANNEL_FL;
+		info->position[1] = SPA_AUDIO_CHANNEL_FR;
+		info->position[2] = SPA_AUDIO_CHANNEL_LFE;
+		break;
+	case 2:
+		info->position[0] = SPA_AUDIO_CHANNEL_FL;
+		info->position[1] = SPA_AUDIO_CHANNEL_FR;
+		break;
+	case 1:
+		info->position[0] = SPA_AUDIO_CHANNEL_MONO;
+		break;
+	default:
+		for (size_t i = 0; i < info->channels; i++) {
+			info->position[i] = SPA_AUDIO_CHANNEL_UNKNOWN;
+		}
+		break;
+	}
+
+    // todo - create a new unconnected pwstream
+    // todo - connect the stream to a sink and source so that it
+    //can be used in the qpw_init_out and qpw_init_in functions
+
+    //todo - unlock the threaded loop.
```

Comment:
I understand that you will complete the function in the next commits, and hence the comments.

But in this commit the function will not compile as it is missing the expected return. Commits should not leave the projects in an inconsistent/broken state, or else reverting or cherry-picking can be very dangerous.